ui/src/treeDataProvider/AzureTreeDataProvider.ts

```diff
@@ -224,7 +224,7 @@ export class AzureTreeDataProvider implements TreeDataProvider<IAzureNode>, Disp
             .map((n: AzureNode) => { return { data: n, description: '', label: n.treeItem.label }; }));
 
         const options: QuickPickOptions = { placeHolder: localize('selectSubscription', 'Select a Subscription') };
-        const result: AzureNode | string = (await this._ui.showQuickPick(picks, options)).data;
+        const result: AzureNode | string = picks.length === 1 ? picks[0].data : (await this._ui.showQuickPick(picks, options)).data;
```

Comment:
Shouldn't this be handled in the ui package?

Reply:
I'm not sure where I could handle this in the ui package.  The SubscriptionNode file itself and the azureaccounts-extension retrieves all of the subscriptions.

I believe that the AzureDataTreeProvider is the only place where we can control the subscriptionNode pick.

Reply:
So you're okay with this behavior here?

Reply:
Ah...   didn't notice.  Good.